webdriver/tests/user_prompts/get_alert_text.py

```diff
@@ -0,0 +1,50 @@
+from tests.support.asserts import assert_error, assert_success
+
+
+def get_dialog_text(session):
+    return session.transport.send("GET", "session/{session_id}/alert/text"
+                                  .format(session_id=session.session_id))
+
+
+# 18.3 Get Alert Text
+
+def test_no_browsing_context(session, create_window):
+    # 18.3 step 1
+    session.window_handle = create_window()
+    session.close()
+
+    response = get_dialog_text(session)
+    assert_error(response, "no such window")
+
+
+def test_no_user_prompt(session):
+    # 18.3 step 2
+    response = get_dialog_text(session)
+    assert_error(response, "no such alert")
+
+
+def test_get_alert_text(session):
+    # 18.3 step 3
+    session.execute_script("window.alert(\"Hello\");")
+    response = get_dialog_text(session)
+    alert_text = response.body["value"]
+    assert alert_text == "Hello"
+    assert_success(response)
+
+
+def test_get_confirm_text(session):
+    # 18.3 step 3
+    session.execute_script("window.confirm(\"Hello\");")
+    response = get_dialog_text(session)
+    alert_text = response.body["value"]
+    assert alert_text == "Hello"
+    assert_success(response)
+
+
+def test_get_prompt_text(session):
+    # 18.3 step 3
+    session.execute_script("window.prompt(\"Enter Your Name: \", \"Federer\");")
+    response = get_dialog_text(session)
+    alert_text = response.body["value"]
+    assert alert_text == "Enter Your Name: "
```

Comment:
I don’t know if this string is trimmed? You should check HTML and the different browsers.

Reply:
I don't think the string gets trimmed. I tested it in firefox, chrome and edge and it does not get trimmed.

Reply:
Smashing.